DataFormats/Detectors/TRD/src/RawData.cxx

```diff
@@ -84,253 +103,265 @@ uint32_t getlinkerrorflag(const HalfCRUHeader& cruhead, const uint32_t link)
-   * https://alicetrd.web.cern.ch/alicetrd/tdp/main.pdf under 17.2.1
-   *
-   *     |07|06|05|04|03|02|01|00|
-   *     -------------------------
-   * qa: |       Q2        | Q1  |  TrackletMCMHeader::pid
-   *     -------------------------
-   *
-   * TDP: This can be one of these fields HPID0/1/2 (=TrackletHCHeader::pid0/1/2) depending on
-   *      the MCM-CPU.
-   *
-   *     |11|10|09|08|07|06|05|04|03|02|01|00|
-   *     -------------------------------------
-   * qb: |     Q0             |      Q1      |  TrackletMCMData::pid
-   *     -------------------------------------
-   *
-   * TDP: This is the LPID field (=TrackletMCMData::pid).
-   *
-   * Q1 is then calculated like this:
-   *
-   *     |06|05|04|03|02|01|00|
-   *     ----------------------
-   * Q1: |qa.Q1|    qb.Q1     |
-   *     ----------------------
-   *
-   **/
-  qb = data->pid;
-  switch (pidindex) {
-    case 0:                   //Q0
-      pid = (qb >> 5) & 0x7f; // 7 bits at the top of all of Q0
-      break;
-    case 1:                                  //Q1
-      pid = ((qa & 0x3) << 5) | (qb & 0x1f); // 2 bits of qa and 5 bits of qb for Q1 .. 7 bits
-      break;
-    case 2:                   //Q2
-      pid = (qa >> 2) & 0x3f; // 6 bits shifted down by bits 2 only taking 6 bits
-      break;
-    default:
-      LOG(warn) << " unknown pid index of : " << pidindex;
-      break;
+  stream << std::endl;
+  stream << "link  errorflags : ";
+  for (int link = 0; link < 15; link++) {
+    stream << link << ":" << std::hex << std::setw(2) << getHalfCRULinkErrorFlag(halfcru, link) << ",";
   }
-  //PID VERSION 2
-  /*
- switch(pidindex) {
-     case 0 : pid=qa&0xffc>>2;break;
-     case 1 : pid=((qa&0x3)<<5)|(qb>>6);break;
-     case 2 : pid=qb&0x3f;break;
-     default : LOG(warn) << " unknown pid index of : " << pidindex;
- }
- */
-  return pid;
+  stream << std::endl;
+  stream << "0x" << std::hex << halfcru.word0 << " 0x" << halfcru.word12[0] << " 0x" << halfcru.word12[1] << " 0x" << halfcru.word3 << " 0x" << halfcru.word47[0] << " 0x" << halfcru.word47[1] << " 0x" << halfcru.word47[2] << " 0x" << halfcru.word47[3] << std::endl;
+  return stream;
 }
 
-uint32_t setHalfCRUHeader(HalfCRUHeader& cruhead, int crurdhversion, int bunchcrossing, int stopbits, int endpoint, int eventtype, int feeid, int cruid)
+//Tracklet HC Header
+
+void constructTrackletHCHeader(TrackletHCHeader& header, int sector, int stack, int layer, int side, int chipclock, int format)
 {
-  cruhead.BunchCrossing = bunchcrossing;
-  cruhead.StopBit = stopbits;
-  cruhead.EndPoint = endpoint;
-  cruhead.EventType = eventtype;
-  cruhead.HeaderVersion = crurdhversion;
-  //cruhead.FeeID = feeid;
-  //cruhead.CRUID = cruid;
-  return 0;
+  header.word = 0;
+
+  header.format = format;
+  header.supermodule = ~sector;
+  header.stack = ~stack;
+  header.layer = ~layer;
+  header.side = ~side;
+
+  header.MCLK = chipclock;
+  header.one = 1;
```

Comment:
According to the TDP, which might be wrong, this fields name is 'header.zero' and is always 0?